lighthouse-core/gather/driver.js

```diff
@@ -143,17 +143,33 @@ class Driver {
         (_ => reject(new Error('The asynchronous expression exceeded the allotted time of 60s'))),
         60000
       );
+
       this.sendCommand('Runtime.evaluate', {
-        expression: asyncExpression,
+        expression: `(function wrapInNativePromise() {
+          const __nativePromise = window.__nativePromise || Promise;
+          return new __nativePromise(function(resolve) {
+            const wrapError = ${wrapRuntimeEvalErrorInBrowser.toString()};
+            try {
+              __nativePromise.resolve(${asyncExpression}).then(resolve, wrapError);
+            } catch (e) {
+              wrapError(e);
+            }
+          });
+        }())`,
         includeCommandLineAPI: true,
         awaitPromise: true,
         returnByValue: true
       }).then(result => {
         clearTimeout(asyncTimeout);
+        const value = result.result.value;
+
         if (result.exceptionDetails) {
-          reject(result.exceptionDetails.exception.value);
+          // An error occurred before we could even enter our try block, should be *very* rare
+          reject(new Error('an unknown driver error occurred'));
+        } if (value && value.__failedInBrowser) {
+          reject(Object.assign(new Error(), value));
         } else {
-          resolve(result.result.value);
+          resolve(value);
```

Comment:
we can save for another issue (maybe this is also #941), but the fact that there are three cases here
- success
- failure due to driver error
- failure (of some sort) due to the evaluated expression

makes it feel like we shouldn't be conflating the last two, but I'm not exactly sure of an elegant way to do this, or even what gatherers should do (catch or re-throw) if we do differentiate by type